pkg/controller/workloaddispatcher/incrementaldispatcher_test.go

```diff
@@ -0,0 +1,289 @@
+		wantErr        error
+		remoteClusters []string
+		clusters       []kueue.MultiKueueCluster
+	}{
+		{
+			name:           "dispatcher name mismatch",
+			dispatcherName: "other",
+			workload:       baseWorkloadBuilder.DeepCopy(),
+		},
+		{
+			name:           "workload not found",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       nil,
+			wantErr:        apierrors.NewNotFound(schema.GroupResource{Group: kueue.GroupVersion.Group, Resource: "workloads"}, workloadName),
+		},
+		{
+			name:           "workload deleted",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.Clone().DeletionTimestamp(now).Finalizers("kubernetes").Obj(),
+		},
+		{
+			name:           "admission check nil",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+		},
+		{
+			name:           "admission check is rejected",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStateRejected,
+			},
+		},
+		{
+			name:           "admission check is ready",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStateReady,
+			},
+		},
+		{
+			name:           "already assigned to cluster",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.Clone().ClusterName("assigned").Obj(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+		},
+		{
+			name:           "workload is already finished",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.Clone().Finished().Obj(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+		{
+			name:           "workload has quota reserved",
+			dispatcherName: kueueconfig.MultiKueueDispatcherModeIncremental,
+			workload:       baseWorkloadBuilder.DeepCopy(),
+			mkAcState: &kueue.AdmissionCheckState{
+				Name:  "ac1",
+				State: kueue.CheckStatePending,
+			},
+			remoteClusters: []string{"cluster1"},
+			clusters: []kueue.MultiKueueCluster{
+				*utiltesting.MakeMultiKueueCluster("cluster1").
+					KubeConfig(kueue.SecretLocationType, "cluster1").
+					Generation(1).
+					Obj(),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			objs := []runtime.Object{}
```

Comment:
```suggestion
			objs := []client.Object{}
```

What about `client.Object` to leverage controller-runtime?